### ISSUE TITLE
Some downloads failing from nhk.or.jp with IndexError

### ISSUE THREAD

**Issue by sjthespian** [ytdl-org/youtube-dl#31028]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

From a Mac:
```
❯ youtube-dl -v -o "Dining with the Chef - S2022E15 %(title)s.%(ext)s" https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2019324/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'-o', u'Dining with the Chef - S2022E15 %(title)s.%(ext)s', u'https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2019324/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 2.7.16 (CPython) - Darwin-20.6.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 5.0.1, ffprobe 5.0.1, rtmpdump 2.4
[debug] Proxy map: {}
[NhkVod] 2019-324: Downloading JSON metadata
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/usr/local/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 475, in main
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 465, in _real_main
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2069, in download
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 808, in extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/nhk.py", line 130, in _real_extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/nhk.py", line 32, in _extract_episode_info
IndexError: list index out of range
```

From Ubuntu 20.04 (python3)
```
❯ youtube-dl -v -o "Dining with the Chef - S2022E15 %(title)s.%(ext)s" https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2019324/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '-o', 'Dining with the Chef - S2022E15 %(title)s.%(ext)s', 'https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2019324/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.8.10 (CPython) - Linux-5.4.0-113-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 5.0.1-1, ffprobe 5.0.1-1
[debug] Proxy map: {}
[NhkVod] 2019-324: Downloading JSON metadata
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 475, in main
  File "/usr/local/bin/youtube-dl/youtube_dl/__init__.py", line 465, in _real_main
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2068, in download
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 808, in extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 815, in wrapper
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 836, in __extract_info
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 534, in extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/nhk.py", line 130, in _real_extract
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/nhk.py", line 31, in _extract_episode_info
IndexError: list index out of range
```

## Description

Downloads are failing on some video files from NHK (nhk.or.jp) but others are working fine. I have tried this from both my Mac and an Ubuntu 20.04 box, the mac is running python 2 (thanks Apple), the Ubuntu box has defaults to python 3. Both throw the same error.

Another failing URL: https://www3.nhk.or.jp/nhkworld/en/ondemand/video/4026164/

This may only be happening with the most recent episode of a series. The URL above fails, but the previous episode of the same show at https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2019323/ works fine. https://www3.nhk.or.jp/nhkworld/en/ondemand/video/4026161/ is another example of a working URL, while the one above (4026164) fails.

You can find their full VOD catalog at https://www3.nhk.or.jp/nhkworld/en/ondemand/latest/?type=tvEpisode&.

**Comment by october262** [comment]
play the video and press F12 on your keyboard to bring up the developer tools,
go to network tab, in the search box, type  chunklist.m3u8, the first hit that shows,
right click and copy link address then just feed the link into youtube-dl to download the video.

**Comment by dirkf** [comment]
Same problem as https://github.com/yt-dlp/yt-dlp/issues/4066.

This patch will be pushed to the git master soon:
```
--- old/youtube_dl/extractor/nhk.py
+++ new/youtube_dl/extractor/nhk.py
@@ -8,7 +8,7 @@ from ..utils import urljoin
 
 
 class NhkBaseIE(InfoExtractor):
-    _API_URL_TEMPLATE = 'https://api.nhk.or.jp/nhkworld/%sod%slist/v7a/%s/%s/%s/all%s.json'
+    _API_URL_TEMPLATE = 'https://nwapi.nhk.jp/nhkworld/%sod%slist/v7b/%s/%s/%s/all%s.json'
     _BASE_URL_REGEX = r'https?://www3\.nhk\.or\.jp/nhkworld/(?P<lang>[a-z]{2})/ondemand'
     _TYPE_REGEX = r'/(?P<type>video|audio)/'
 
```

**Comment by sjthespian** [comment]
Thanks for the patch! The workaround is roughly the same thing I do with This Old House since it seems to be hit or miss as to how much they have obscured the URL on any given episode.

**Comment by dirkf** [comment]
"This Old House" is supposed to be a supported site, with URLs like https://www.thisoldhouse.com/how-to/how-to-build-storage-bench (various other patterns after the domain are handled):
```
$ python test/test_download.py TestDownload.test_ThisOldHouse
[ThisOldHouse] how-to-build-storage-bench: Downloading webpage
[Zype] 5dcdddf673c3f956ef5db202: Downloading JSON metadata
[Zype] 5dcdddf673c3f956ef5db202: Downloading m3u8 information
[info] Writing video description metadata as JSON to: test_ThisOldHouse_5dcdddf673c3f956ef5db202.info.json
.
----------------------------------------------------------------------
Ran 1 test in 9.488s

OK
$
```
Feel free to raise a broken site support issue if a TOH page isn't being properly handled.

**Comment by dirkf** [comment]
Fixed in 8a158a936c8b002ef536e9e2b778ded02c09c0fa.

**Comment by sjthespian** [comment]
@dirkf I will have to see if the TOH issue reoccurs when they post a new episode. It may be something specific to their authenticated "Insiders" pages. I just checked, and the non-protected new videos do download correctly. I have opened a couple of issues for them in the past - they like to change their URLs and break things. :)
